org_fedora_oscap/model.py

```diff
@@ -0,0 +1,281 @@
+        self.oval = ""
+        self.tailoring = ""
+        self.archive = ""
+        self.verified = ""
+        self.root = pathlib.Path(root)
+
+    def record_verification(self, path):
+        assert path in self.labelled_files
+        self.verified = path
+
+    def add_content_archive(self, fname):
+        path = pathlib.Path(fname)
+        self.labelled_files[path] = None
+        self.archive = path
+
+    def _assign_content_type(self, attribute_name, new_value):
+        old_value = getattr(self, attribute_name)
+        if old_value:
+            msg = (
+                f"When dealing with {attribute_name}, "
+                f"there was already the {old_value.name} when setting the new {new_value.name}")
+            raise RuntimeError(msg)
+        setattr(self, attribute_name, new_value)
+
+    def add_file(self, fname, label):
+        path = pathlib.Path(fname)
+        if label == content_handling.CONTENT_TYPES["TAILORING"]:
+            self._assign_content_type("tailoring", path)
+        elif label == content_handling.CONTENT_TYPES["DATASTREAM"]:
+            self._assign_content_type("datastream", path)
+        elif label == content_handling.CONTENT_TYPES["OVAL"]:
+            self._assign_content_type("oval", path)
+        elif label == content_handling.CONTENT_TYPES["XCCDF_CHECKLIST"]:
+            self._assign_content_type("xccdf", path)
+        self.labelled_files[path] = label
+
+    def _datastream_content(self):
+        if not self.datastream:
+            return None
+        if not self.datastream.exists():
+            return None
+        return self.datastream
+
+    def _xccdf_content(self):
+        if not self.xccdf or not self.oval:
+            return None
+        if not (self.xccdf.exists() and self.oval.exists()):
+            return None
+        return self.xccdf
+
+    def find_expected_usable_content(self, relative_expected_content_path):
+        content_path = self.root / relative_expected_content_path
+        elligible_main_content = (self._datastream_content(), self._xccdf_content())
+
+        if content_path in elligible_main_content:
+            return content_path
+        else:
+            if not content_path.exists():
+                msg = f"Couldn't find '{content_path}' among the available content"
+            else:
+                msg = (
+                    "File '{content_path}' is not a valid datastream "
+                    "or a valid XCCDF of a XCCDF-OVAL file tuple")
+            raise content_handling.ContentHandlingError(msg)
+
+    def select_main_usable_content(self):
+        elligible_main_content = (self._datastream_content(), self._xccdf_content())
+        if not any(elligible_main_content):
+            msg = (
+                "Couldn't find a valid datastream or a valid XCCDF-OVAL file tuple "
+                "among the available content")
+            raise content_handling.ContentHandlingError(msg)
+        if elligible_main_content[0]:
```

Comment:
this perhaps require a comment or better unpacking the tuple on line  into 2 variables that would have good names and changing the condition.

Reply:
or one can avoid the tuple completely.